Marching cubes

coat-soup opened the pull request:
Managed to get marching cubes working with actual simulation data. It's on a super small scale right now because cell lookups are calculated for every voxel. Next step is chunking to get this possible on a real scale.

<img width="803" height="769" alt="image" src="https://github.com/user-attachments/assets/e85f49e2-ad26-4d4c-9592-966fd1de2fd8" />

RaulPPelaez reviewed (commented):
I think your current implementation is going to give you trouble in the future, but you got it working and its a pretty good baseline. These pregenerated tables are going to get in the way.
Also, you know I am on a crusade to try to keep your repo clean. So I must point out the .vs folder, and the lib folder, which seem to contain garbage. There is also a seemingly unnecessary log.txt file.

coat-soup commented:
> Also, you know I am on a crusade to try to keep your repo clean. So I must point out the .vs folder, and the lib folder, which seem to contain garbage. There is also a seemingly unnecessary log.txt file.

Good idea. I have removed the .vs folder and added it the gitgnore. Thanks for catching that. I've also removed the log file, but the library folder is part of an editor plugin that was updated when I first built the C# solution, so I'm hesitant to remove it as it's third party software I don't want to break.

RaulPPelaez commented:
Look up some common .gitignore for godot projects. You have 181 files changed, many of them seem like build artifacts, libs and the likes. For instance, the official one seems to be https://github.com/github/gitignore/blob/main/Godot.gitignore

Anyhow, I will trust your judgement on this front from now on :P
